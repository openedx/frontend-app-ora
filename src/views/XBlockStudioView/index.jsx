import React, { useEffect } from 'react';

import XBlockStudioViewProvider from './components/XBlockStudioViewProvider';
import StudioSchedule from './components/StudioSchedule';
import StudioViewSteps from './components/StudioViewSteps';
import StudioViewSettings from './components/StudioViewSettings';
import StudioViewRubric from './components/StudioViewRubric';
import StudioViewTitle from './components/StudioViewTitle';
import StudioViewPrompt from './components/StudioViewPrompt';

import './index.scss';

export const XBlockStudioView = () => {
  useEffect(() => {
    if (window.parent.length > 0) {
      new ResizeObserver(() => {
        window.parent.postMessage({ type: 'plugin.resize', payload: { height: document.body.scrollHeight } }, document.referrer);
      }).observe(document.body);
    }
  }, []);

  return (
    <XBlockStudioViewProvider>
      <div id="ora-xblock-studio-view">
        <StudioViewTitle />
        <StudioViewPrompt />
        <StudioSchedule />
        <StudioViewSteps />
        <StudioViewSettings />
        <StudioViewRubric />
      </div>
    </XBlockStudioViewProvider>
  );
};

export default XBlockStudioView;
