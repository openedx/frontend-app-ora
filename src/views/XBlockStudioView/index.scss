@import "@edx/paragon/scss/core/core";

#ora-xblock-studio-view {
  max-width: 1024px;
  display: flex;
  flex-direction: column;
  gap: map-get($spacers, 3);

  @include media-breakpoint-down(md) {
    padding-left: 40px;
    padding-right: 40px;
  }


  .block-title {
    white-space: nowrap;
    display: flex;
    justify-content: space-between;

    h2 {
      overflow: hidden;
      text-overflow: ellipsis;
    }
  }
}
