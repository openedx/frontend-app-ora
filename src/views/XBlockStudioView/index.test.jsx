import { shallow } from '@edx/react-unit-test-utils';

import XBlockStudioView from './index';

jest.mock('./components/XBlockStudioViewProvider', () => 'XBlockStudioViewProvider');
jest.mock('./components/StudioSchedule', () => 'StudioSchedule');
jest.mock('./components/StudioViewSteps', () => 'StudioViewSteps');
jest.mock('./components/StudioViewSettings', () => 'StudioViewSettings');
jest.mock('./components/StudioViewRubric', () => 'StudioViewRubric');
jest.mock('./components/StudioViewTitle', () => 'StudioViewTitle');
jest.mock('./components/StudioViewPrompt', () => 'StudioViewPrompt');

describe('<XBlockStudioView />', () => {
  it('should render', () => {
    const wrapper = shallow(<XBlockStudioView />);
    expect(wrapper.snapshot).toMatchSnapshot();

    expect(wrapper.instance.findByType('XBlockStudioViewProvider').length).toBe(1);
    expect(wrapper.instance.findByType('StudioSchedule').length).toBe(1);
    expect(wrapper.instance.findByType('StudioViewSteps').length).toBe(1);
    expect(wrapper.instance.findByType('StudioViewSettings').length).toBe(1);
    expect(wrapper.instance.findByType('StudioViewRubric').length).toBe(1);
    expect(wrapper.instance.findByType('StudioViewTitle').length).toBe(1);
    expect(wrapper.instance.findByType('StudioViewPrompt').length).toBe(1);
  });
});
