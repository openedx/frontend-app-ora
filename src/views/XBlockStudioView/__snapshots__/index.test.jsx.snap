// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<XBlockStudioView /> should render 1`] = `
<XBlockStudioViewProvider>
  <div
    id="ora-xblock-studio-view"
  >
    <StudioViewTitle />
    <StudioViewPrompt />
    <StudioSchedule />
    <StudioViewSteps />
    <StudioViewSettings />
    <StudioViewRubric />
  </div>
</XBlockStudioViewProvider>
`;
